Lib/test/test_netrc.py

```diff
@@ -1,45 +1,147 @@
-import netrc, os, unittest, sys, textwrap
-from test.support import os_helper
+import netrc, os, unittest, sys, textwrap, tempfile
+
+from test import support
+from contextlib import ExitStack
 
 try:
     import pwd
 except ImportError:
     pwd = None
 
-temp_filename = os_helper.TESTFN
+
+class NetrcEnvironment:
+    """
+    Context manager for setting up an isolated environment to test `.netrc` file handling.
+
+    This class configures a temporary directory for the `.netrc` file and environment variables, providing
+    a controlled setup to simulate different scenarios.
+    """
+
+    def __enter__(self) -> 'NetrcEnvironment':
+        """
+        Enters the managed environment.
+        """
+        self.stack = ExitStack()
+        self.environ = self.stack.enter_context(support.os_helper.EnvironmentVarGuard())
+        self.tmpdir = self.stack.enter_context(tempfile.TemporaryDirectory())
```

Comment:
Avoid using `tempfile.TemporaryDirectory` but prefer using a test-only temporary directory via `test.support.os_helper.temp_dir`

Reply:
Why was this rollbacked?